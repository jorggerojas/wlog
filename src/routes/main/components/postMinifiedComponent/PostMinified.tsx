import React from "react";
import { Link, BrowserRouter as Router } from "react-router-dom";

interface PostProps {
  title: string;
  date: string;
  user: string;
  summary: string;
  badge: string;
  id: string;
}

const PostMinified = ({ title, date, user, summary, badge, id }: PostProps) => {
  return (
    <div>
      <div className="uk-card-large  uk-card uk-card-default uk-card-hover">
        <div className="uk-card-header">
          <h3 className="uk-card-title">{title.toUpperCase()}</h3>
          <div className="uk-card-badge uk-label uk-text-uppercase">
            {badge}
          </div>
          <p className="uk-text-meta uk-text-italic uk-margin-remove-top">
            <time>Created on {date}</time> by{" "}
            <Router>
              <Link to="" className="uk-button-text">
                {user.toUpperCase()}
              </Link>
            </Router>
          </p>
        </div>
        <div className="uk-card-body uk-text-break">
          <p className="uk-text-italic">
            {summary.length > 200 ? summary.slice(0, 150) + "..." : summary}
          </p>
          <a href="#s" className=" uk-button uk-button-text">
            Read more...
          </a>
        </div>
      </div>
    </div>
  );
};

export default PostMinified;
