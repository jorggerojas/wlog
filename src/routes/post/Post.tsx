import React, { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";
import Header from "../main/components/Header";
import axios from "axios";
import cookie from "react-cookies";
import { parseDate, URL } from "../../config";
import { useFormik } from "formik";
import * as yup from "yup";

const Post = () => {
  const { user, id } = JSON.parse(JSON.stringify(useParams()));
  const [keywordList, setKeywordList] = useState([]);
  const [updateInfo, setUpdateInfo] = useState(false);
  const [inputs, setInputs] = useState(false);
  const [data, setData] = useState({
    id: "",
    title: "",
    summary: "",
    dateLog: "2020-07-19",
    username: "",
    content: "",
    keywords: [],
    comments: [],
  });
  useEffect(() => {
    getPostInfo().then((response: any) => {
      cookie.load("ROLE") === "ADMIN" || user === response.data.username
        ? setUpdateInfo(true)
        : setUpdateInfo(false);
      setData(response.data);
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data]);
  let {
    title,
    summary,
    dateLog,
    username,
    content,
    keywords /*comments*/,
  } = data;
  const getPostInfo = () => {
    return axios.get(`${URL}/users/${user}/posts/${id}`);
  };
  return (
    <div className="uk-animation-fade">
      <Header />
      <div className="uk-padding">
        <h1 className="uk-text-left uk-text-bold uk-text-italic">
          {title}{" "}
          {updateInfo ? (
            <span>
              {inputs ? (
                <span
                  uk-icon="close"
                  style={{ cursor: "pointer", color: "red" }}
                  onClick={() => setInputs(false)}
                ></span>
              ) : (
                <span
                  uk-icon="pencil"
                  style={{ cursor: "pointer", color: "#0F7AE5" }}
                  onClick={() => setInputs(true)}
                ></span>
              )}{" "}
              <span
                className={`${inputs ? "uk-hidden" : "uk-visible"}`}
                uk-icon="trash"
                style={{ cursor: "pointer", color: "red" }}
              ></span>
            </span>
          ) : null}
        </h1>
        <p className="uk-text-meta uk-text-light">
          Created by:{" "}
          <span>
            <Link to={`/user/${user}`}>{user.toUpperCase()}</Link>
          </span>{" "}
          on {""}
          {parseDate(dateLog)}
        </p>
        {/* cc */}
        <form className="post uk-align-center">
          <ul uk-accordion="multiple: true">
            {inputs ? (
              <li className="uk-open">
                <Link className="uk-accordion-title" to="#s">
                  <legend className="uk-legend">Title</legend>
                </Link>
                <div className="uk-accordion-content">
                  <div className="uk-margin">
                    <input
                      className="uk-input"
                      type="text"
                      placeholder="Something cool..."
                    />
                  </div>
                </div>
              </li>
            ) : null}
            <li uk-accordion="multiple:true">
              <Link className="uk-accordion-title" to="#">
                <legend className="uk-legend">Summary</legend>
              </Link>
              <div className="uk-accordion-content">
                <div className="uk-margin uk-padding-small">
                  {inputs ? (
                    <textarea
                      className="uk-textarea"
                      rows={3}
                      style={{ resize: "none" }}
                      placeholder="Something short and cool"
                    />
                  ) : (
                    summary
                  )}
                </div>
              </div>
            </li>
            <li uk-accordion="multiple:true">
              <Link className="uk-accordion-title" to="#">
                <legend className="uk-legend">Content</legend>
              </Link>
              <div className="uk-accordion-content">
                <div className="uk-margin uk-padding-small">
                  {inputs ? (
                    <textarea
                      className="uk-textarea"
                      rows={10}
                      placeholder="All you need is type"
                      style={{ resize: "none" }}
                    />
                  ) : (
                    content
                  )}
                </div>
              </div>
            </li>
            <li uk-accordion="multiple:true">
              <Link className="uk-accordion-title" to="#">
                <legend className="uk-legend">Keywords</legend>
              </Link>
              <div className="uk-accordion-content">
                <div className="uk-margin uk-padding-small">
                  {inputs ? (
                    <div>
                      <p className="uk-text-meta uk-text-small">
                        Press the key{" "}
                        <span className="uk-text-italic">"Enter"</span> ,{" "}
                        <span className="uk-text-italic">"Space"</span> ,{" "}
                        <span className="uk-text-italic">"." (period)</span> or{" "}
                        <span className="uk-text-italic">"," (comma)</span> for
                        add the word to the list.
                      </p>
                      <input
                        className="uk-input"
                        type="text"
                        placeholder="Pocas e importantes"
                        defaultValue={""}
                      />
                      <div className="uk-placeholder uk-text-center">
                        {keywords.map((keyword: string) => {
                          return (
                            <span
                              key={keyword}
                              className="uk-margin-small-right uk-badge uk-padding-small"
                            >
                              {keyword.toUpperCase()}
                            </span>
                          );
                        })}
                      </div>
                    </div>
                  ) : (
                    <div className="uk-text-center">
                      {keywords.map((keyword: string) => {
                        return (
                          <span
                            key={keyword}
                            className="uk-margin-small-right uk-badge uk-padding-small"
                          >
                            {keyword.toUpperCase()}
                          </span>
                        );
                      })}
                    </div>
                  )}
                </div>
              </div>
            </li>
          </ul>
          {inputs ? (
            <div className="uk-margin uk-padding-small uk-text-right">
              <input
                className="uk-button uk-button-primary"
                type="submit"
                value="Save"
              />
            </div>
          ) : null}
        </form>
        {/* jjj */}
        {inputs ? (
          ""
        ) : (
          <div>
            <div className="uk-margin-bottom">
              <article
                className="uk-comment uk-comment-primary"
                style={{ minHeight: "13rem !important" }}
              >
                <header
                  className="uk-comment-header uk-grid uk-flex-middle"
                  uk-grid={""}
                >
                  <div className="uk-width-1-2@s uk-grid" uk-grid="">
                    <div className="uk-width-1-1@s">
                      <h4 className="uk-comment-title ">
                        Write a comment for this post
                      </h4>
                    </div>
                    <div className="uk-width-1-1@s uk-margin-bottom">
                      <p className=" uk-text-meta uk-margin">
                        Remember be kind with the author and all the community
                      </p>
                    </div>
                  </div>
                </header>
                <div className="uk-comment-body uk-margin-bottom">
                  <div className="uk-margin">
                    <textarea
                      className="uk-textarea"
                      rows={3}
                      style={{ resize: "none" }}
                      defaultValue={""}
                      placeholder="What do you think?"
                    ></textarea>
                  </div>
                  <div className="uk-align-right">
                    <button className="uk-button uk-button-primary">
                      Comment
                    </button>
                  </div>
                </div>
              </article>
            </div>
            <div>
              <article className="uk-comment uk-comment-primary uk-margin">
                <header
                  className="uk-comment-header uk-grid-medium uk-flex-middle"
                  uk-grid=""
                >
                  <div className="uk-width-expand">
                    <p
                      style={{ fontSize: "1rem" }}
                      className="uk-article-title uk-button uk-button-text"
                    >
                      <Link className="uk-link-reset" to={`/user/${username}`}>
                        {username}
                      </Link>
                    </p>
                    {inputs ? (
                      <div className="uk-align-right">
                        <span
                          style={{ cursor: "pointer" }}
                          uk-icon="icon:close"
                        />
                      </div>
                    ) : null}
                    <ul className="uk-comment-meta uk-subnav uk-subnav-divider uk-margin-remove-top">
                      <li className="uk-text-italic">{"dateLog"}</li>
                    </ul>
                  </div>
                </header>
                <div className="uk-comment-body">
                  <p>Jajajajaja bien cool</p>
                </div>
              </article>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};
export default Post;
