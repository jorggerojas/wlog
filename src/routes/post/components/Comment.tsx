// @flow
import React from "react";
import cookie from "react-cookies";
import { Link } from "react-router-dom";
import { parseDate } from "../../../config";

interface CommentProps {
  comment: string;
  user: string;
  date: string;
  id: string;
}

const Comment = ({ comment, user, date, id }: CommentProps) => {
  return (
    <article className="uk-comment uk-comment-primary uk-margin">
      <header
        className="uk-comment-header uk-grid-medium uk-flex-middle"
        uk-grid=""
      >
        <div className="uk-width-expand">
          <p
            style={{ fontSize: "1rem" }}
            className="uk-article-title uk-button uk-button-text"
          >
            <Link className="uk-link-reset" to={`/user/${user}`}>
              {user}
            </Link>
          </p>
          {cookie.load("USER") === user ? (
            <div className="uk-align-right">
              <span
                style={{ cursor: "pointer" }}
                uk-icon="icon:close"
                onClick={() => console.log(id)}
              />
            </div>
          ) : null}
          <ul className="uk-comment-meta uk-subnav uk-text-left uk-margin-remove-top">
            <li className="uk-text-italic uk-padding-remove">
              {parseDate(date)}
            </li>
          </ul>
        </div>
      </header>
      <div className="uk-comment-body">
        <p className="uk-text-italic">{`${comment
          .toUpperCase()
          .substr(0, 1)}${comment
          .toLowerCase()
          .substr(1, comment.length - 1)}`}</p>
      </div>
    </article>
  );
};

export default Comment;
