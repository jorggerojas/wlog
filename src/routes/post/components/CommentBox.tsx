// @flow
import React, { useState } from "react";
import { useFormik } from "formik";
import { object, string } from "yup";
import t from "typy";
import axios from "axios";
import cookie from "react-cookies";
import swal from "sweetalert";
import { URL } from "../../../config";
import { Form } from "../../../styles/styles";

interface CommentBoxProps {
  post: string;
  setComments: Function;
  getComments: Function;
}

const CommentBox = ({ post, setComments, getComments }: CommentBoxProps) => {
  const { USER, ROLE } = cookie.loadAll();
  const [disabled, setDisabled] = useState(false);
  const formik = useFormik({
    initialValues: {
      comment: "",
    },
    validationSchema: object({
      comment: string().required(
        "Comment is required if you want to say something"
      ),
    }),
    onSubmit: (values) => {
      if (t(ROLE).isNullOrUndefined) {
        swal({
          title: "Wait!",
          text:
            "You need to have an account to comment in this or another post",
          buttons: {
            cancel: {
              visible: true,
              text: "Cancel",
            },
            ok: {
              visible: true,
              text: "Let me in!",
            },
          },
        }).then((willSign) => {
          if (willSign) {
            window.location.href = "/sign";
          } else return;
        });
      } else {
        setDisabled(true);
        axios({
          method: "POST",
          url: `${URL}/users/${USER}/posts/${post}/comments`,
          headers: {
            Authorization: `Bearer ${cookie.load("TOKEN")}`,
          },
          data: {
            user: USER,
            post: post,
            content: values.comment,
          },
        })
          .then(() => {
            formik.values.comment = "";
            getComments().then(({ data }: any) => {
              setComments(data.content);
              setTimeout(() => {
                setDisabled(false);
              }, 500);
            });
          })
          .catch(() => {
            swal("You can't comment this post right now, try later");
          });
      }
    },
  });
  return (
    <Form
      onSubmit={formik.handleSubmit}
      className="uk-comment uk-comment-primary"
    >
      <header className="uk-comment-header uk-grid uk-flex-middle" uk-grid={""}>
        <div className="uk-width-1-2@s uk-grid" uk-grid="">
          <div className="uk-width-1-1@s">
            <h4 className="uk-comment-title ">Write a comment for this post</h4>
          </div>
          <div className="uk-width-1-1@s uk-margin-bottom">
            <p className=" uk-text-meta uk-margin">
              Remember be kind with the author and all the community
            </p>
          </div>
        </div>
      </header>
      <div className="uk-comment-body uk-margin-bottom">
        <div className="uk-margin">
          <textarea
            className="uk-textarea"
            rows={3}
            placeholder="What do you think?"
            {...formik.getFieldProps("comment")}
          />
        </div>
        {t(formik.touched.comment).safeObject &&
        t(formik.errors.comment).safeObject ? (
          <div className="uk-text-danger uk-text-bold">
            {formik.errors.comment}
          </div>
        ) : null}
        <div className="uk-align-right">
          {t(disabled).isTrue ? (
            <div uk-spinner="ratio:1.2" className="divSalmon"></div>
          ) : (
            <input
              type="submit"
              disabled={disabled}
              value="Comment"
              className="uk-button uk-button-primary"
            />
          )}
        </div>
      </div>
    </Form>
  );
};

export default CommentBox;
