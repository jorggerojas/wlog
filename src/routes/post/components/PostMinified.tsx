// @flow
import React from "react";
import { Link, useParams } from "react-router-dom";
import { LinkUserComment, SpanBadge, Title3 } from "../../../styles/text";
import { CardBody, DivBadge } from "../../../styles/containers";

interface PostProps {
  title: string;
  date: string;
  user?: string;
  summary: string;
  badge: string;
  large?: boolean;
  id: string;
}

const PostMinified = ({
  title,
  date,
  user,
  summary,
  badge,
  id,
  large,
}: PostProps) => {
  const { username } = useParams<{ username: string }>();
  return (
    <div>
      <CardBody
        className={`uk-card-${
          large ? "large" : "small"
        }  uk-card uk-card-default uk-card-hover`}
      >
        <div className="uk-card-header uk-margin-remove-bottom">
          <Title3 className="uk-card-title uk-margin-remove-bottom">
            <Link
              to={`/user/${user ?? username}/post/${id}`}
              className="uk-link-heading uk-link-reset"
            >
              {title.length > 17
                ? title.toUpperCase().substr(0, 14) + "..."
                : title.toUpperCase()}
            </Link>
          </Title3>
          {user ? (
            <DivBadge className="uk-card-badge uk-label uk-text-uppercase uk-visible@s">
              {badge}
            </DivBadge>
          ) : null}
          <div
            className={`${
              !user ? "" : "uk-hidden@s"
            } uk-margin-small uk-margin-remove-top`}
          >
            <SpanBadge className="uk-badge">{badge.toUpperCase()}</SpanBadge>
          </div>
          {user ? (
            <p className="uk-text-meta uk-text-italic uk-margin-remove-top">
              <time>{date}</time> by{" "}
              <Link to={`/user/${user}`} className="uk-button-text">
                {user.toUpperCase()}
              </Link>
            </p>
          ) : (
            <p className="uk-text-meta uk-text-italic uk-margin-remove-top">
              <time>{date}</time>
            </p>
          )}
        </div>
        <div className="uk-padding uk-text-break">
          <p className="uk-text-italic  uk-visible@s">
            {summary.length > 200 ? summary.slice(0, 150) + "..." : summary}
          </p>
          <LinkUserComment
            href={`/user/${user ?? username}/post/${id}`}
            className=" uk-button uk-button-text"
          >
            Read more...
          </LinkUserComment>
        </div>
      </CardBody>
    </div>
  );
};

export default PostMinified;
