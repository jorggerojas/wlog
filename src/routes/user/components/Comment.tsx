// @flow
import React from "react";
import { Link } from "react-router-dom";

interface CommentProps {
  comment: string;
  date: string;
  post: string;
  id: string;
}

const Comment = ({ comment, date, post, id }: CommentProps) => {
  return (
    <div className="uk-card-small uk-card-default uk-card-hover">
      <div className="uk-card-body">
        <span className="uk-text-meta uk-text-italic">{date}</span>
        <p className="uk-text-secondary uk-text-italic">"{comment}"</p>
      </div>
      <div className="uk-card-footer uk-text-right">
        <Link to={`/post/${id}`} className="uk-link-heading">
          Ver Post{" "}
        </Link>
      </div>
    </div>
  );
};

export default Comment;
