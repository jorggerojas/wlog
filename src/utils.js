import React from "react";
import { render, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import "@testing-library/jest-dom";

const renderWithRouter = (ui, { route = "/", ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  const finishLoading = () =>
    waitFor(() =>
      expect(utils.queryByText("Loading, please wait...")).toBeNull()
    );
  return {
    ...utils,
    finishLoading,
    history,
  };
};

export {
  Simulate,
  wait,
  render,
  cleanup,
  renderIntoDocument,
  fireEvent,
} from "@testing-library/react";
export { renderWithRouter };

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...render(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history,
  };
}
