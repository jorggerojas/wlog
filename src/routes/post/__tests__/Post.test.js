import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import axiosMock from "axios";
import { Route } from "react-router-dom";
import Post from "../Post";
import Comment from "../components/Comment";
import CommentBox from "../components/CommentBox";
import PostMinified from "../components/PostMinified";
import { renderWithRouter, renderWithProviders } from "../../../utils";

afterEach(() => {
  cleanup();
  localStorage.clear();
});
jest.mock("axios");

describe("<Post/> tree", () => {
  test("render component", async () => {
    const post = {
      title: "My super real post",
      date: "2020-12-12",
      user: "RealUser1",
      summary: "The summary of the post",
      content: "The content of the post",
      badge: ["WAR", "TEST"],
      id: "1",
    };
    axiosMock.get.mockImplementation(() =>
      Promise.resolve({
        data: post,
      })
    );
    localStorage.setItem("ROLE", "ADMIN");
    localStorage.setItem("USER", post.user);
    localStorage.setItem("content", JSON.stringify(post));
    const { container } = renderWithProviders(
      <Route path="/user/:user/post/:id">
        <Post handle={() => {}} theme={true} />
      </Route>,
      {
        route: "/user/real/post/1",
      }
    );
    await expect(container).toHaveTextContent(post.user.toUpperCase());
    await expect(container).toMatchSnapshot();
  });
  describe("children components", () => {
    test("<CommentBox/> component render", async () => {
      const { container, getByTestId } = render(
        <CommentBox getComments={() => {}} setComments={() => {}} post={1} />
      );
      const box = getByTestId("comment");
      fireEvent.change(box, { target: { value: "My comment" } });
      await expect(container.innerHTML).toMatch(
        "Remember be kind with the author and all the community"
      );
      await expect(box.innerHTML).toMatch("My comment");
      await expect(container).toMatchSnapshot();
    });
    describe("<Comment/> component", () => {
      test("Without close icon", async () => {
        const props = {
          comment: "A real comment made by MamberroiArcos",
          user: "MamberroiArcos",
          date: "2020-12-30",
          id: "myId",
          post: "aRealPost",
        };
        const { container } = render(<Comment {...props} />);
        await expect(container.innerHTML).toMatch(
          `${props.comment
            .toUpperCase()
            .substr(0, 1)}${props.comment
            .toLowerCase()
            .substr(1, props.comment.length - 1)}`
        );
        await expect(container.innerHTML).not.toMatch("close");
        await expect(container).toMatchSnapshot();
      });
      test("With close icon", async () => {
        localStorage.setItem("USER", "RealUser1");
        localStorage.setItem("ROLE", "ADMIN");
        const props = {
          comment: "A real comment made by MamberroiArcos",
          user: "MamberroiArcos",
          date: "2020-12-30",
          id: "myId",
          post: "aRealPost",
        };
        const { container } = render(<Comment {...props} />);
        await expect(container.innerHTML).toMatch(
          `${props.comment
            .toUpperCase()
            .substr(0, 1)}${props.comment
            .toLowerCase()
            .substr(1, props.comment.length - 1)}`
        );
        await expect(
          container.innerHTML.includes(`uk-icon="icon:close"`)
        ).toBeTruthy();
        await expect(container).toMatchSnapshot();
      });
    });
  });
});

describe("<PostMinified/> component", () => {
  test("render component", async () => {
    const post = {
      title: "My super real post",
      date: "2020-12-12",
      user: "RealUser1",
      summary: "The summary of the post",
      content: "The content of the post",
      badge: "WAR",
      id: "1",
      large: true,
    };
    const { container } = renderWithRouter(<PostMinified {...post} />);
    await expect(container.innerHTML).toMatch(
      `${post.title.toUpperCase().substr(0, 14)}...`
    );
    await expect(container.innerHTML).toMatch(`${post.badge}`);
    await expect(container).toMatchSnapshot();
  });
});
