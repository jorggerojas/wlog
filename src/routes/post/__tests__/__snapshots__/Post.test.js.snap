// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/> tree children components <Comment/> component With close icon 1`] = `
<div>
  <article
    class="sc-fubCfw kajVyM uk-comment uk-comment-primary uk-margin"
  >
    <header
      class="uk-comment-header uk-grid-medium uk-flex-middle"
      uk-grid=""
    >
      <div
        class="uk-width-expand"
      >
        <p
          class="sc-kLgntA gwUKQS uk-article-title uk-button uk-button-text uk-text-bold"
        >
          <a
            class="sc-hBEYos hfgEZq"
            href="/user/MamberroiArcos"
          >
            MamberroiArcos
          </a>
        </p>
        <div
          class="uk-align-right"
        >
          <span
            class="pointer"
            uk-icon="icon:close"
          />
        </div>
        <ul
          class="uk-comment-meta uk-subnav uk-text-left uk-margin-remove-top"
        >
          <li
            class="sc-gWHgXt hGGwwD uk-padding-remove"
          >
            December 30th of 2020
          </li>
        </ul>
      </div>
    </header>
    <div
      class="uk-comment-body"
    >
      <p
        class="sc-cBNfnY bdrpve uk-text-italic"
      >
        A real comment made by mamberroiarcos
      </p>
    </div>
  </article>
</div>
`;

exports[`<Post/> tree children components <Comment/> component Without close icon 1`] = `
<div>
  <article
    class="sc-fubCfw kajVyM uk-comment uk-comment-primary uk-margin"
  >
    <header
      class="uk-comment-header uk-grid-medium uk-flex-middle"
      uk-grid=""
    >
      <div
        class="uk-width-expand"
      >
        <p
          class="sc-kLgntA gwUKQS uk-article-title uk-button uk-button-text uk-text-bold"
        >
          <a
            class="sc-hBEYos hfgEZq"
            href="/user/MamberroiArcos"
          >
            MamberroiArcos
          </a>
        </p>
        <ul
          class="uk-comment-meta uk-subnav uk-text-left uk-margin-remove-top"
        >
          <li
            class="sc-gWHgXt hGGwwD uk-padding-remove"
          >
            December 30th of 2020
          </li>
        </ul>
      </div>
    </header>
    <div
      class="uk-comment-body"
    >
      <p
        class="sc-cBNfnY bdrpve uk-text-italic"
      >
        A real comment made by mamberroiarcos
      </p>
    </div>
  </article>
</div>
`;

exports[`<Post/> tree children components <CommentBox/> component render 1`] = `
<div>
  <form
    class="sc-kEjbxe biPcz uk-comment uk-comment-primary"
  >
    <header
      class="uk-comment-header uk-grid uk-flex-middle"
      uk-grid=""
    >
      <div
        class="uk-width-1-2@s uk-grid"
        uk-grid=""
      >
        <div
          class="uk-width-1-1@s"
        >
          <h4
            class="sc-iktFzd clEUXB uk-comment-title "
          >
            Write a comment for this post
          </h4>
        </div>
        <div
          class="uk-width-1-1@s uk-margin-bottom"
        >
          <p
            class="sc-lmoMRL cZQxry  uk-text-meta uk-margin"
          >
            Remember be kind with the author and all the community
          </p>
        </div>
      </div>
    </header>
    <div
      class="uk-comment-body uk-margin-bottom"
    >
      <div
        class="uk-margin"
      >
        <textarea
          class="sc-ezrdKe zScU uk-textarea"
          data-testid="comment"
          name="comment"
          placeholder="What do you think?"
          rows="3"
        >
          My comment
        </textarea>
      </div>
      <div
        class="uk-align-right"
      >
        <input
          class="sc-hiSbYr iiDUxc uk-button uk-button-primary uk-margin-small-top"
          type="submit"
          value="Comment"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`<Post/> tree render component 1`] = `
<div>
  <div
    class="sc-eCssSg csSspB uk-animation-fade"
  >
    <div
      uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky"
    >
      <nav
        class="sc-hKgILt kSSoBK uk-navbar uk-navbar-container uk-margin"
      >
        <div
          class="uk-navbar-left"
        >
          <a
            class="uk-navbar-item uk-logo"
            href="/"
          >
            <span
              class="sc-fFubgz iOMCry"
            >
              Wlog
            </span>
          </a>
        </div>
        <div
          class="uk-navbar-right"
        >
          <ul
            class="uk-navbar-nav"
          >
            <li
              class="uk-visible"
            >
              <div
                class="mid"
              >
                <label
                  class="rocker rocker-small"
                >
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <span
                    class="switch-left"
                  >
                    ðŸŒš
                  </span>
                  <span
                    class="switch-right"
                  >
                    ðŸŒž
                  </span>
                </label>
              </div>
            </li>
            <li
              class="uk-visible@s"
            >
              <a
                href="/user/RealUser1"
              >
                <span
                  class="sc-fFubgz iOMCry"
                >
                  REALUSER1
                </span>
              </a>
            </li>
            <li
              class="uk-hidden@s"
            >
              <a
                href="/user/RealUser1"
              >
                <span
                  class="sc-fFubgz iOMCry"
                >
                  REALUSER1
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="uk-navbar-center"
        >
          <a
            class="uk-navbar-toggle"
            href="/create"
          >
            <span
              class="white"
              uk-icon="icon:plus-circle"
            />
          </a>
        </div>
      </nav>
    </div>
    <div
      class="uk-padding"
    >
      <div>
        <h1
          class="sc-iwyYcG iwahXu uk-text-left uk-text-bold uk-text-italic"
        >
          My super real post
           
          <span>
            <span
              class="pointer blue"
              uk-icon="pencil"
            />
             
            <span
              class="uk-visible pointer red"
              uk-icon="trash"
            />
          </span>
        </h1>
        <p
          class="sc-lmoMRL cZQxry uk-text-meta uk-text-light"
        >
          Created by 
          
          <a
            class="sc-kstrdz iEkqgo"
            href="/user/real"
          >
            REAL
          </a>
           on 
          
          December 12th of 2020
        </p>
        <form
          class="post uk-align-center"
        >
          <ul
            uk-accordion="multiple: true"
          >
            <li
              uk-accordion="multiple:true"
            >
              <a
                class="uk-accordion-title"
                href="/user/real/post/1"
              >
                <legend
                  class="sc-fodVxV fwBrnF"
                >
                  Summary
                </legend>
              </a>
              <div
                class="uk-accordion-content"
              >
                <div
                  class="uk-padding-small"
                >
                  <p
                    class="sc-lmoMRL cZQxry"
                  >
                    The summary of the post
                  </p>
                </div>
              </div>
            </li>
            <li
              uk-accordion="multiple:true"
            >
              <a
                class="uk-accordion-title"
                href="/user/real/post/1"
              >
                <legend
                  class="sc-fodVxV fwBrnF"
                >
                  Content
                </legend>
              </a>
              <div
                class="uk-accordion-content"
              >
                <div
                  class="uk-padding-small"
                >
                  <p
                    class="sc-lmoMRL cZQxry"
                  >
                    The content of the post
                  </p>
                </div>
              </div>
            </li>
            <li
              uk-accordion="multiple:true"
            >
              <a
                class="uk-accordion-title"
                href="/user/real/post/1"
              >
                <legend
                  class="sc-fodVxV fwBrnF"
                >
                  Keywords
                </legend>
              </a>
              <div
                class="uk-accordion-content"
              >
                <div
                  class=" uk-padding-small"
                >
                  <div
                    class="uk-text-center"
                  />
                </div>
              </div>
            </li>
          </ul>
        </form>
        <div>
          <div
            class="uk-margin-bottom"
          >
            <form
              class="sc-kEjbxe biPcz uk-comment uk-comment-primary"
            >
              <header
                class="uk-comment-header uk-grid uk-flex-middle"
                uk-grid=""
              >
                <div
                  class="uk-width-1-2@s uk-grid"
                  uk-grid=""
                >
                  <div
                    class="uk-width-1-1@s"
                  >
                    <h4
                      class="sc-iktFzd clEUXB uk-comment-title "
                    >
                      Write a comment for this post
                    </h4>
                  </div>
                  <div
                    class="uk-width-1-1@s uk-margin-bottom"
                  >
                    <p
                      class="sc-lmoMRL cZQxry  uk-text-meta uk-margin"
                    >
                      Remember be kind with the author and all the community
                    </p>
                  </div>
                </div>
              </header>
              <div
                class="uk-comment-body uk-margin-bottom"
              >
                <div
                  class="uk-margin"
                >
                  <textarea
                    class="sc-ezrdKe zScU uk-textarea"
                    data-testid="comment"
                    name="comment"
                    placeholder="What do you think?"
                    rows="3"
                  />
                </div>
                <div
                  class="uk-align-right"
                >
                  <input
                    class="sc-hiSbYr iiDUxc uk-button uk-button-primary uk-margin-small-top"
                    type="submit"
                    value="Comment"
                  />
                </div>
              </div>
            </form>
          </div>
          <div
            class="sc-iqHYGH igejut uk-section uk-section-default uk-padding-small"
          >
            <h4
              class="sc-iktFzd clEUXB uk-text-center uk-text-italic"
            >
              No comments in this post
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PostMinified/> component render component 1`] = `
<div>
  <div>
    <div
      class="sc-crrsfI fcNBva uk-card-large  uk-card uk-card-default uk-card-hover"
    >
      <div
        class="uk-card-header uk-margin-remove-bottom"
      >
        <h3
          class="sc-idOhPF hBzjfg uk-card-title uk-margin-remove-bottom"
        >
          <a
            class="uk-link-heading uk-link-reset"
            href="/user/RealUser1/post/1"
          >
            MY SUPER REAL ...
          </a>
        </h3>
        <div
          class="sc-gKsewC jbSaSR uk-card-badge uk-label uk-text-uppercase uk-visible@s"
        >
          WAR
        </div>
        <div
          class="uk-hidden@s uk-margin-small uk-margin-remove-top"
        >
          <span
            class="sc-bkzZxe gsWPPD uk-badge"
          >
            WAR
          </span>
        </div>
        <p
          class="uk-text-meta uk-text-italic uk-margin-remove-top"
        >
          <time>
            2020-12-12
          </time>
           by
           
          <a
            class="uk-button-text"
            href="/user/RealUser1"
          >
            REALUSER1
          </a>
        </p>
      </div>
      <div
        class="uk-padding uk-text-break"
      >
        <p
          class="uk-text-italic  uk-visible@s"
        >
          The summary of the post
        </p>
        <a
          class="sc-hBEYos hfgEZq  uk-button uk-button-text"
          href="/user/RealUser1/post/1"
        >
          Read more...
        </a>
      </div>
    </div>
  </div>
</div>
`;
